firebase/admin/firestore.rules

```diff
@@ -281,6 +281,12 @@ service cloud.firestore {
 
       allow update: if isCurrentAdmin() || editingMyData();
 
+      match /externalData/{dataSource} {
+        allow read: if canReadExistingUser();
+        allow create: if canCreateUser();
+        allow update: if isCurrentAdmin() || editingMyData();
+      }
+
```

Comment:
I think this approach is going in the right direction. I agree that we want the same access properties as for the parent `/users/{roarUid}` document. But I have two issues (one minor and one major, I think):

- (The major one): I believe the functions `canReadExistingUser()`, `canCreateUser()` and `isCurrentAdmin()` ultimately rely on inspecting `resource.data`. For the `/users/{roarUid}` documents, `resource.data` will contain the stuff we want. But once we are in `/users/{roarUid}/externalData/{dataSource}`, then `resource.data` will point to that document and the functions above will not work as expected. So I believe we need to create versions of these functions that rely on database reads of the parent document. I used a similar approach [here](https://github.com/yeatmanlab/roar-firekit/blob/3daebd98c84c5d6a2924b77945bc87b326309998/firebase/assessment/firestore.rules#L315).
- (The minor one): I'm wondering if we even want to concern ourselves with `allow create` and `allow update` here. We could set them both to `if false` under the assumption that all external data is populated in cloud functions. Then we only need to allow reads. If we decide to adopt this, then we could delete the `updateUserExternalData` method of `RoarFirekit`. What do you think?

Lastly, in reviewing the code above, I think I made an error in the definiteion of `isCurrentAdmin()` on Line 274. Shouldn't we also include `isStudyAdmin()`? That is, perhaps it should read
```
function isCurrentAdmin() {
        return keysNotUpdated(readOnlyKeys().concat(userUpdateKeys())) && (isEduAdmin() || isFamilyAdmin() || isStudyAdmin());
      }
```

Can you double check my reasoning there?